lib/core/Observer.ts

```diff
@@ -1,12 +1,15 @@
 import IBlockchain from './interfaces/IBlockchain';
+import IOperationRateLimiter from './interfaces/IOperationRateLimiter';
 import IOperationStore from './interfaces/IOperationStore';
 import ITransactionProcessor from './interfaces/ITransactionProcessor';
 import ITransactionStore from './interfaces/ITransactionStore';
 import IUnresolvableTransactionStore from './interfaces/IUnresolvableTransactionStore';
 import IVersionManager from './interfaces/IVersionManager';
+import OperationRateLimiter from './versions/latest/OperationRateLimiter';
 import SharedErrorCode from '../common/SharedErrorCode';
 import timeSpan = require('time-span');
 import TransactionModel from '../common/models/TransactionModel';
+import TransactionNumber from '../bitcoin/TransactionNumber';
```

Comment:
Design: Similar to the previous comment, I have yet to see why this is needed, but block height is intentionally hidden from the core up until this point. If this class must be used, at very lease it needs to be moved to `common` folder.

Reply:
Actually now that you drew my attention to this, is this calculation specific to bitcoin? If so, we shouldn't use it.